ersilia/core/tracking.py

```diff
@@ -27,6 +44,7 @@ def write_persistent_file(contents):
         with open(PERSISTENT_FILE_PATH, "a") as f:
             f.write(f"{contents}\n")
 
+    log_files_metrics(TEMP_FILE_LOGS)
```

Comment:
Can you actually move this into `close_persistent_file` instead? I know I said it should be in this method, but I think it makes more sense to only run when we're closing the file instead.